DirectProgramming/DPC++FPGA/Tutorials/Features/experimental/register_map_and_streaming_interfaces/README.md

```diff
@@ -0,0 +1,324 @@
+Intel® oneAPI Toolkits, which provides system checks to find missing
+dependencies and permissions errors.
+[Learn more](https://software.intel.com/content/www/us/en/develop/documentation/diagnostic-utility-user-guide/top.html).
+
+ ### In Third-Party Integrated Development Environments (IDEs)
+
+You can compile and run this tutorial in the Eclipse* IDE (in Linux*) and the Visual Studio* IDE (in Windows*). For instructions, refer to the following link: [Intel® oneAPI DPC++ FPGA Workflows on Third-Party IDEs]([https://software.intel.com/en-us/articles/intel-oneapi-dpcpp-fpga-workflow-on-ide](https://www.intel.com/content/www/us/en/developer/articles/technical/intel-oneapi-dpcpp-fpga-workflow-on-ide.html))
+
+## Examining the Reports
+
+Locate `report.html` in the `register_map_and_streaming_interfaces_report.prj/reports/` directory. Open the report in any of the following web browsers:  Chrome*, Firefox*, Edge*, or Internet Explorer*.
+
+Open the **Views** menu and select **System Viewer**.
+
+In the left-hand pane, select **FunctorRegisterMapControlIP** or **LambdaRegisterMapControlIP** under the System hierarchy.
+
+In the main **System Viewer** pane, the kernel control interface and kernel argument interfaces are shown. They show that the `start`, `busy` and `done` kernel control interfaces are implemented in register map interfaces, the `arg_input` and `arg_output` kernel arguments are implemented in register map interfaces. The `arg_n` kernel argument is implemented in streaming interface in the **FunctorRegisterMapControlIP**, and in register map interface in the **LambdaRegisterMapControlIP**.
+
+Similarly, in the left-hand pane, select **StreamingControlIP** or **LambdaStreamingControlIP** under the System hierarchy.
+
+In the main **System Viewer** pane, the kernel control interface and kernel argument interfaces are shown. They show that the `start`, `done`, `ready_in` and `ready_out` kernel control interfaces are implemented in streaming interfaces, and the `arg_input`, `arg_output` and `arg_n` kernel arguments are all implemented in streaming interfaces.
+
+## Running the Sample
+
+ 1. Run the sample on the FPGA emulator (the kernel executes on the CPU):
+     ```
+     ./register_map_and_streaming_interfaces.fpga_emu     (Linux)
+     register_map_and_streaming_interfaces.fpga_emu.exe   (Windows)
+     ```
+2. Run the sample on the FPGA simulator:
+     ```
+     ./register_map_and_streaming_interfaces.fpga_sim      (Linux)
+     register_map_and_streaming_interfaces.fpga_sim.exe        (Windows)
+     ```
+
+### Example of Output
+
+```
+Running the kernel with streaming control implemented in the functor programming model
+	 Done
+
+Running the kernel with register map control implemented in the functor programming model
+	 Done
+
+Running kernel with streaming control implemented in the lambda programming model
+	 Done
+
+Running kernel with register map control implemented in the lambda programming model
+	 Done
+
+PASSED
+```
```

Comment:
Please add a note and screenshot of looking at the waveforms in Modelsim. Seeing and understanding the signals in the Modelsim GUI is important to RTL authors.

Reply:
If we attach some modelsim screenshots here, we might need to test them in our regtests to ensure the screenshot here doesn't get out of sync with the product. After discussing with Bain, that might be doable but may be quite a lot of work. What do you think about having some hand-drawn waveform diagrams (similar to the ones in HLS docs) instead of real modelsim screenshots? Thanks!

Reply:
hand-drawn waveforms would be fine too.

Reply:
Added hand-drawn waveforms at the bottom of the README. I'm intentionally ambiguous about the specific register addresses in the description so that the README don't need to be constantly updated.